nixos/modules/services/mail/public-inbox.nix

```diff
@@ -0,0 +1,569 @@
+            description = "URL where this inbox can be accessed over HTTP.";
+          };
+          options.description = mkOption {
+            type = types.str;
+            example = "user/dev discussion of public-inbox itself";
+            description = "User-visible description for the repository.";
+          };
+          options.newsgroup = mkOption {
+            type = with types; nullOr str;
+            default = null;
+            description = "NNTP group name for the inbox.";
+          };
+          options.watch = mkOption {
+            type = with types; listOf str;
+            default = [];
+            description = "Paths for ${manref "public-inbox-watch" 1} to monitor for new mail.";
+            example = [ "maildir:/path/to/test.example.com.git" ];
+          };
+          options.watchheader = mkOption {
+            type = with types; nullOr str;
+            default = null;
+            example = "List-Id:<test@example.com>";
+            description = ''
+              If specified, ${manref "public-inbox-watch" 1} will only process
+              mail containing a matching header.
+            '';
+          };
+          options.coderepo = mkOption {
+            type = (types.listOf (types.enum (attrNames cfg.settings.coderepo))) // {
+              description = "list of coderepo names";
+            };
+            default = [];
+            description = "Nicknames of a 'coderepo' section associated with the inbox.";
+          };
+        }));
+      };
+    };
+    mda = {
+      enable = mkEnableOption "the public-inbox Mail Delivery Agent";
+      args = mkOption {
+        type = with types; listOf str;
+        default = [];
+        description = "Command-line arguments to pass to ${manref "public-inbox-mda" 1}.";
+      };
+    };
+    http = {
+      enable = mkEnableOption "the public-inbox HTTP server";
+      mounts = mkOption {
+        type = with types; listOf str;
+        default = [ "/" ];
+        example = [ "/lists/archives" ];
+        description = ''
+          Root paths or URLs that public-inbox will be served on.
+          If domain parts are present, only requests to those
+          domains will be accepted.
+        '';
+      };
+      args = mkOption {
+        type = with types; listOf str;
+        default = ["-W0"];
+        description = "Command-line arguments to pass to ${manref "public-inbox-httpd" 1}.";
+      };
+      port = mkOption {
+        type = with types; nullOr (either str port);
+        default = 80;
+        example = "/run/public-inbox-httpd.sock";
+        description = ''
+          Listening port or systemd's ListenStream= entry
+          to be used as a reverse proxy, eg. in nginx:
+          <code>locations."/inbox".proxyPass = "http://unix:''${config.services.public-inbox.http.port}:/inbox";</code>
+          Set to null and use <code>systemd.sockets.public-inbox-httpd.listenStreams</code>
+          if you need a more advanced listening.
```

Comment:
I don't think the extra complexity of taking two different kinds of values here is worth the small convenience of being able to open the firewall automatically.

Reply:
I'm not sure to understand what you want, isn't `openFirewall` working well enough?